apple/testing/default_runner/ios_xctestrun_runner.bzl

```diff
@@ -0,0 +1,163 @@
+"""
+A iOS test runner rule that uses xctestrun files to run unit test bundles on
+simulators. This rule currently doesn't support UI tests or running on device.
+"""
+
+load("@build_bazel_rules_apple//apple/testing:apple_test_rules.bzl", "AppleTestRunnerInfo")
+
+def _get_template_substitutions(
+        *,
+        device_type,
+        os_version,
+        simulator_creator,
+        random,
+        xctestrun_template):
+    substitutions = {
+        "device_type": device_type,
+        "os_version": os_version,
+        "simulator_creator.py": simulator_creator,
+        # "ordered" isn't a special string, but anything besides "random" for this field runs in order
+        "test_order": "random" if random else "ordered",
+        "xctestrun_template": xctestrun_template,
+    }
+
+    return {"%({})s".format(key): value for key, value in substitutions.items()}
+
+def _get_execution_environment(ctx):
+    xcode_version = str(ctx.attr._xcode_config[apple_common.XcodeVersionConfig].xcode_version())
+    if not xcode_version:
+        fail("error: No xcode_version in _xcode_config")
+
+    return {"XCODE_VERSION_OVERRIDE": xcode_version}
+
+def _ios_xctestrun_runner_impl(ctx):
+    os_version = str(ctx.attr.os_version or ctx.fragments.objc.ios_simulator_version or
+                     ctx.attr._xcode_config[apple_common.XcodeProperties].default_ios_sdk_version)
+
+    # TODO: Ideally we would be smarter about picking a device, but we don't know what the versoin of Xcode supports
+    device_type = ctx.attr.device_type or ctx.fragments.objc.ios_simulator_device or "iPhone 12"
+
+    if not os_version:
+        fail("error: os_version must be set on ios_xctestrun_runner, or passed with --ios_simulator_version")
+    if not device_type:
+        fail("error: device_type must be set on ios_xctestrun_runner, or passed with --ios_simulator_device")
```

Comment:
Do we still want the "iPhone 12" default?

Reply:
Seems both of these checks can't currently be hit.

Reply:
we have to have some default, i picked one that should be supported by xcode 12 and 13+. the os_version check supposedly can, although i think that could only happen if you provided your own, potentially invalid xcode version config file. the device_type one can't, but I figured we should leave it in case we break it?